### ISSUE TITLE
tslm causes an error in dwtest

### ISSUE THREAD

**Issue by robjhyndman** [robjhyndman/forecast#329]
MRE:

```
library(fpp)
dwtest(lm(sales ~ advert, data=advert))
dwtest(tslm(sales ~ advert, data=advert))
```

These should be identical.

**Comment by mitchelloharawild** [comment]
As far as I can tell, it's an simple bug, but difficult problem.

dwtest checks if fit$x is a matrix (which is the case for tslm, but not for lm). Not sure if fit$x is used anywhere else, but it may be worth renaming to be more obscure.

```R
library(fpp)
dwtest(lm(sales ~ advert, data=advert))
fit <- tslm(sales ~ advert, data=advert)
fit$x <- NULL
dwtest(fit)
```

These are identical.

**Comment by mitchelloharawild** [comment]
Hi Rob,

Revisiting this, I've made the changes in #333 again and ran R CMD check with no errors.
Do you recall any particular problems that this caused (leading to the revert 85cf2d8c746420616f122a3c9a21759a010f5593) other than the Travis CI errors?

**Comment by robjhyndman** [comment]
The problem is that the dwtests are not equivalent as noted above. The revert was because of the error noted in my 1 June comment at https://github.com/robjhyndman/forecast/pull/333. The current version doesn't cause an error, but the dwtests are not equivalent.

**Comment by mitchelloharawild** [comment]
I believe that error was fixed in #334, as that code no longer errors for me.
I'll add that code snippet to the "test-tslm.R", and it still passes --as-cran checks.

If that's sufficient, I've made the commit and PR.

Thanks for the clarification.
